### PR TITLE
Add models package and Search result list feature

### PR DESCRIPTION
**Pull request by akserg**
- Add models package and move models and dtos there
- Add Avatar, Checkbox, List, ListItem, ListItemAvatar, ListItemButton, ListItemIcon and ListItemText to ui package
- Add Search list and item features
- Add Search route and replace map on it

Screenshot 2023-11-18 at 20 51 19

### CODE REVIEW

File: packages/models/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "@samovar/models",
+    "version": "1.0.0",
+    "main": "./dist/index.js",
```
Comment (arcadiy-magomedov) [inline packages/models/package.json]: must have "private": true

otherwise monorepo won't work

File: packages/models/dist/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+declare class User {
+    id: string;
```
Comment (arcadiy-magomedov) [inline packages/models/dist/index.d.ts]: we do not need dist folder, see how ui package is done

File: packages/models/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    ],
+    "scripts": {
+      "build": "tsup src/index.ts --format esm,cjs --dts --external node"
+    },
+    "dependencies": {
+      "class-validator": "^0.14.0"
+    },
+    "devDependencies": {
+      "tsup": "^6.2.3",
+      "typescript": "^5.1.3"
```
Comment (arcadiy-magomedov) [inline packages/models/package.json]: typescript already installed in the root package

File: packages/models/package.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      "dist/**/*"
+    ],
+    "scripts": {
+      "build": "tsup src/index.ts --format esm,cjs --dts --external node"
+    },
+    "dependencies": {
+      "class-validator": "^0.14.0"
+    },
+    "devDependencies": {
+      "tsup": "^6.2.3",
```
Comment (arcadiy-magomedov) [inline packages/models/package.json]: no bundling needed

File: packages/models/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "extends": "tsconfig/dtos.json",
```
Comment (arcadiy-magomedov) [inline packages/models/tsconfig.json]: it should not work, you have not added "tsconfig": "*" to the devDependencies

File: packages/models/dist/index.d.ts (hunk 2 above)
Comment (akserg) [reply packages/models/dist/index.d.ts]: ok

File: packages/models/package.json (hunk 1 above)
Comment (akserg) [reply packages/models/package.json]: Added

File: packages/models/dist/index.d.ts (hunk 2 above)
Comment (akserg) [reply packages/models/dist/index.d.ts]: Removed

File: packages/models/package.json (hunk 3 above)
Comment (akserg) [reply packages/models/package.json]: Removed

File: packages/models/package.json (hunk 4 above)
Comment (akserg) [reply packages/models/package.json]: Removed

File: packages/models/tsconfig.json (hunk 5 above)
Comment (akserg) [reply packages/models/tsconfig.json]: Added

File: packages/models/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "extends": "tsconfig/models-library.json",
+  "include": ["."],
+  "exclude": ["dist", "build", "node_modules"]
```
Comment (magom001) [inline packages/models/tsconfig.json:4]: dist and build do not exist any more

**Review by arcadiy-magomedov — APPROVED** [review]

**Pull request merged**